v03_pipeline/lib/methods/sex_check.py

```diff
@@ -0,0 +1,207 @@
+    return mt.cols()
+
+
+def generate_fstat_plot(
+    ht: hl.Table,
+    xy_fstat_threshold: float,
+    xx_fstat_threshold: float,
+) -> io.BytesIO:
+    # Plot histogram of fstat values
+    # Returns the plot saved as a binary buffer.
+    buf = io.BytesIO()
+    df = ht.to_pandas()
+    plt.clf()
+    plt.hist(df['f_stat'])
+    plt.xlabel('Fstat')
+    plt.ylabel('Frequency')
+    plt.axvline(xy_fstat_threshold, color='blue', linestyle='dashed', linewidth=1)
+    plt.axvline(xx_fstat_threshold, color='red', linestyle='dashed', linewidth=1)
+    plt.savefig(buf, format='png')
+    buf.seek(0)
+    return buf
+
+
+def call_sex(  # noqa: PLR0913
+    mt: hl.MatrixTable,
+    reference_genome: ReferenceGenome,
+    use_chrY_cov: bool = False,  # noqa: N803
+    chrY_cov_threshold: float = 0.1,  # noqa: N803
+    normalization_contig: str = 'chr20',
+    xy_fstat_threshold: float = 0.75,
+    xx_fstat_threshold: float = 0.5,
+    aaf_threshold: float = 0.05,
+    call_rate_threshold: float = 0.25,
+) -> tuple[hl.Table, io.BytesIO]:
+    """
+    Call sex for the samples in a given callset and export results file to the desired path.
+
+    :param mt: MatrixTable containing samples with chrY variants
+    :param reference_genome: ReferenceGenome, either GRCh37 or GRCh38
+    :param use_chrY_cov: Set to True to calculate and use chrY coverage for sex inference.
+        Will also compute and report chrX coverages.
+        Default is False
+    :param chrY_cov_threshold: Y coverage threshold used to infer sex aneuploidies.
+        XY samples below and XX samples above this threshold will be inferred as having aneuploidies.
+        Default is 0.1
+    :param normalization_contig: Chosen chromosome for calculating normalized coverage. Default is "chr20"
+    :param xy_fstat_threshold: F-stat threshold above which a sample will be called XY. Default is 0.75
+    :param xx_fstat_threshold: F-stat threshold below which a sample will be called XX. Default is 0.5
+    :param aaf_threshold: Alternate allele frequency threshold for `hl.impute_sex`. Default is 0.05
+    :param call_rate_threshold: Minimum required call rate. Default is 0.25
+    :return Table with imputed sex annotations, and the fstat plot.
+    """
+
+    valid_contigs = {*reference_genome.autosomes, *reference_genome.sex_chromosomes}
+    if normalization_contig not in valid_contigs:
+        msg = f'Contig: {normalization_contig} is invalid for this reference genome'
+        raise ValueError(msg)
+
+    # Filter to SNVs and biallelics
+    # NB: We should already have filtered biallelics, but just
+    mt = mt.filter_rows(hl.is_snp(mt.alleles[0], mt.alleles[1]))
+
+    # Filter to PASS variants only (variants with empty or missing filter set)
+    # TODO: Make this an optional argument before moving to gnomad_methods
+    mt = mt.filter_rows(
+        hl.is_missing(mt.filters) | (mt.filters.length() == 0),
+        keep=True,
+    )
+
+    impute_sex_ht = hl.impute_sex(
```

Comment:
One note here, I removed the `run_hails_impute_sex` function and the line that was filtering to the x config itself.  Per the [hail docs](https://hail.is/docs/0.2/methods/genetics.html#hail.methods.impute_sex) for this method, that filtering is happening within the method itself.  We were also missing the step to "Remove loci in the pseudoautosomal region" anyways.